<a href="https://colab.research.google.com/github/spasatel13/DS-Unit-1-Sprint-2-Statistics/blob/master/module4-bayesian/Philipp_Libenson_W2_D4_Bayesian_Inference_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




### 1) In your own words, summarize the difference between Bayesian and Frequentist statistics.

**Your answer here.** \
Frequentist states that there is a a single number for mean and variance in a population, even though we could never know it. He uses the sample data to estimate them. The Bayesian thinks that there is a range for both mean and variance the estimated range of which can be updated with each successive experiment. 

###For questions 2) - 8) use the following information:

Let A be the event of surviving the sinking of the Titanic and B be the event of being a 1st class passenger.

In [1]:
import pandas as pd
import numpy as np


data_url = 'https://raw.githubusercontent.com/Chelsea-Myers/Lambda-Intro/master/Titanic.csv'

df = pd.read_csv(data_url, skipinitialspace=True, header=0)

print(df.shape)
df.head()

(887, 8)


Survived  Pclass  ... Parents/Children_Aboard     Fare
0         0       3  ...                       0   7.2500
1         1       1  ...                       0  71.2833
2         1       3  ...                       0   7.9250
3         1       1  ...                       0  53.1000
4         0       3  ...                       0   8.0500

[5 rows x 8 columns]

In [2]:
#Crosstab of Sex and Survived
pd.crosstab(df['Sex'],df['Survived'],margins=True)

Survived    0    1  All
Sex                    
female     81  233  314
male      464  109  573
All       545  342  887

In [11]:
pd.crosstab(df['Sex'],df['Survived'],normalize="index", margins=True)

Survived         0         1
Sex                         
female    0.257962  0.742038
male      0.809773  0.190227
All       0.614431  0.385569

2) Calculate P(A) = P(Surving)

In [9]:
### YOUR CODE HERE
#The total number of passengers who survived.
total_survived=df['Survived'][df['Survived']==1].count()
# Total on Board
total_onBoard=df['Survived'].count()
P_of_A=total_survived/total_onBoard
P_of_A

0.3855693348365276

3) Calculate P(B) = P(Being a 1st class passenger)

In [13]:
### YOUR CODE HERE
#total passengers in 1st class
total_1stClass = df['Pclass'][df['Pclass']==1].count()
P_of_B=total_1stClass/total_onBoard
P_of_B

0.24351747463359638

4) Calculate P(A $\cap$ B) = P(Being a 1st class passenger and surviving)

In [20]:
pd.crosstab(df['Pclass'],df['Survived'],margins=True)

Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         368  119  487
All       545  342  887

In [31]:
pd.crosstab(df['Pclass'],df['Survived'],margins=True, normalize='index')

Survived         0         1
Pclass                      
1         0.370370  0.629630
2         0.527174  0.472826
3         0.755647  0.244353
All       0.614431  0.385569

In [19]:
### YOUR CODE HERE
condition=(df['Survived']==1)&(df['Pclass']==1)
total_1stClass_andSurvived=len(df[condition])
total_1stClass_andSurvived
P_AandB=total_1stClass_andSurvived/total_onBoard
P_AandB

0.15332581736189402

5) Calculate P(A|B) = P(Surviving given that the person was a 1st class passenger)

In [42]:
pd.crosstab(df['Pclass'],df['Survived'],margins=True, normalize='index')

Survived         0         1
Pclass                      
1         0.370370  0.629630
2         0.527174  0.472826
3         0.755647  0.244353
All       0.614431  0.385569

In [43]:
### YOUR CODE HERE
P_AgivenB=pd.crosstab(df['Pclass'],df['Survived'],margins=True, normalize='index').iloc[0][1]
P_AgivenB

0.6296296296296297

6) Calculate (B|A) = P(Being a first class passenger given the person survived)

In [36]:
pd.crosstab(df['Survived'],df['Pclass'],margins=True, normalize='index')

Pclass           1         2         3
Survived                              
0         0.146789  0.177982  0.675229
1         0.397661  0.254386  0.347953
All       0.243517  0.207441  0.549042

In [39]:
### YOUR CODE HERE
P_BgivenA=pd.crosstab(df['Survived'],df['Pclass'],margins=True, normalize='index').iloc[1][1]
P_BgivenA

0.39766081871345027

7) Show that 

$$P(A|B) = \frac{P(A \cap B)}{P(B)}$$

with the numbers you calculate above.

In [40]:
### YOUR CODE HERE
P_AgivenB==P_AandB/P_of_B

True

8) Show that 

$$P(A|B) = \frac{P(B|A)P(A)}{P(B)}$$

with the numbers you calculated above.

In [44]:
### YOUR CODE HERE
P_AgivenB==P_AgivenB*P_of_B/P_of_B

True

###Use the following information to answer questions 9) - 11) 

Assume that the passengers on the Titanic are representative of all individuals traveling from Western Europe to the United States in the early 20th century.

Let's use the Titanic data sample to estimate the mean fare paid by all individuals traveling from Western Euorpe to the United States during that time.

9) Use a 95% frequentist CI to estimate the mean fare paid by individuals traveling from Western Europe to the US in the early 20th century.

Interpret your confidence interval.

In [48]:
### YOUR CODE HERE
from scipy.stats import t
mean_fare = df['Fare'].mean()
sd_fare = df['Fare'].std() 
n_fare = df['Fare'].count()
se_fare = sd_fare / (np.sqrt(n_fare))

t.interval(alpha= 0.95, df=n_fare - 1, loc=mean_fare, scale=se_fare)

(29.02482490325031, 35.586015457516254)

Your answer here. \
The 95% frequentist CI for the mean fare paid by individuals traveling from Western Europe to the US in the early 20th century is between  29.02 and 35.39

10) Use a 95% Bayesian CI to determine the distribution of the fare paid by individuals traveling from Western Europe to the US in the early 20th century.  Interpret your results

In [49]:
### YOUR CODE HERE
#Using Bayesian MVS (Mean, variance, standard deviation)

from scipy import stats

stats.bayes_mvs(df['Fare'], alpha=0.95)

(Mean(statistic=32.30542018038331, minmax=(29.024824903250337, 35.58601545751628)),
 Variance(statistic=2483.858450355689, minmax=(2262.747559034253, 2726.2430731483255)),
 Std_dev(statistic=49.824230556814946, minmax=(47.568346187714496, 52.213437668365835)))

**Your answer here**. \
We are 95% confident that the true mean of the fair paid by all travelers from Western Europe can be described by a statistical distribution that has a mean of 32.31 dollars and a standard deviation of 49.82 dollars

#Portfolio Project Assignment

Watch the Porfolio Project websites mini-lecture: https://www.youtube.com/watch?v=gxbNCGERUjc&feature=youtu.be&ab_channel=LambdaSchool


Begin to set up your own Portfolio Project website.